src/middleware/builtin.ts

```diff
@@ -74,107 +74,92 @@ export const onlyEvents: Middleware<AnyMiddlewareArgs & { event?: SlackEvent }>
 };
 
 /**
- * Middleware that filters out any event that isn't a view_submission
+ * Middleware that filters out any event that isn't a view_submission or view_closed event
  */
-export const onlyViewSubmits: Middleware<AnyMiddlewareArgs & { view?: ViewSubmitAction }> = ({ view, next }) => {
-  // Filter out anything that isn't a view_submission
-  if (view === undefined) {
-    return;
-  }
-
-  // It matches so we should continue down this middleware listener chain
-  next();
-};
-
-// TODO: is there a way to consolidate the next two methods without too much type refactoring?
-export function matchCallbackId(
-    callbackId: string | RegExp,
-): Middleware<SlackViewMiddlewareArgs> {
-  return ({ payload, next, context }) => {
-    let tempMatches: RegExpMatchArray | null;
-
-    if (!isCallbackIdentifiedBody(payload)) {
+export const onlyViewActions: Middleware<AnyMiddlewareArgs &
+  { view?: (ViewSubmitAction | ViewClosedAction) }> = ({ view, next }) => {
+    // Filter out anything that doesn't have a view
+    if (view === undefined) {
       return;
     }
-    if (typeof callbackId === 'string') {
-      if (payload.callback_id !== callbackId) {
-        return;
-      }
-    } else {
-      tempMatches = payload.callback_id.match(callbackId);
-
-      if (tempMatches !== null) {
-        context['callbackIdMatches'] = tempMatches;
-      } else {
-        return;
-      }
-    }
 
+    // It matches so we should continue down this middleware listener chain
     next();
   };
-}
 
 /**
  * Middleware that checks for matches given constraints
  */
 export function matchConstraints(
```

Comment:
it seems like this should actually be two different functions: one for action constraints and one for view constraints. the code smell here is that when you call this, you always know whether you have one or the other, and within the function body you basically have two non-overlapping branches to handle them.

Reply:
This was done to address https://github.com/slackapi/bolt/issues/266. Do we not think that's valuable anymore?

Reply:
I agree that the only real overlap right now is the callbackId section.